src/cicero/dom.cto

```diff
@@ -146,9 +147,29 @@ asset List extends Section {
 }
 
 /**
- * An inline function can be included in a contract to perform basic calculations
+ * Ergo inlined in contracts or clauses
+ */
+abstract asset Ergo extends Element {
+}
+
+/**
+ * Inline Ergo declarations (constants, functions) can be included in a contract to perform basic calculations
  */
 @scope(ContractTemplate)
-asset Function extends Element {
+asset Declarations extends Ergo {
   o String code
-}
\ No newline at end of file
+}
+
+/**
+ * Inline Ergo expressions can be included to construct part of the output contract or check constraints on parameters
+ */
+abstract asset Expression extends Ergo {
+  o String code
+}
+
+@scope(ContractTemplate, ClauseTemplate)
+asset ElementExpression extends Expression{ // An expression to build part of the contract. Should return an Element
```

Comment:
Where do you imagine this appearing in the contract?

Reply:
Here is an example with a payment schedule inside the contract (Notional Amount Schedule in https://www.sec.gov/Archives/edgar/data/1508478/000119312511077213/dex1021.htm). I would imagine you would not want to create that table by hand, but you would want to do it based on other parameters in the contract (e.g., interest date, schedule).

A simpler example might be just calculating a total amount based on a principal and interest inside the contract itself.